examples/pipelineruns/pipelinerun-with-pipelinespec-and-taskspec.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineRun
+metadata:
+  name: pipelinerun-with-taskspec-to-echo-greetings
+spec:
+  pipelineSpec:
+    params:
+      - name: GREETINGS
+        description: "Greetings, default is Hello World!"
+        type: string
+        default: "Hello World!"
+    tasks:
+      - name: echo-good-morning
+        taskSpec:
+          inputs:
+            params:
+              - name: MESSAGE
+                type: string
+                default: "Hello World!"
+          steps:
+            - name: echo
+              image: ubuntu
+              command:
+                - echo
+              args:
+                - "$(inputs.params.MESSAGE)"
```

Comment:
hey @afrittoli `$(params.GREETINGS)` will not work here as task is aware of only one input param named `MESSAGE`, here in this example, `GREETINGS`'s value is assigned to task's input param `MESSAGE`. If we replace line 26 with `$(params.GREETINGS)`, pod will have a output of:

```
$(params.GREETINGS)
```

Reply:
I wonder if it should work or not though… My gut feeling as a user would be that I can use pipeline params in the embedded `TaskSpec` (which is not the case with this PR), right ?

Reply:
I guess if we wanted that we'd need syntax like `pipeline.params.FOO`, right?  Since both the pipeline and the task could have a param called FOO and they might have different intended meanings.  So `$(params.FOO)` would be a bit confusing (to Tekton controller, but also to users).  Or we could assume that task params always "win" if there's a same named param in both pipeline and task.

Reply:
> I guess if we wanted that we'd need syntax like `pipeline.params.FOO`, right? Since both the pipeline and the task could have a param called FOO and they might have different intended meanings. So `$(params.FOO)` would be a bit confusing (to Tekton controller, but also to users). Or we could assume that task params always "win" if there's a same named param in both pipeline and task.

Not entirely sure to be honest. I think the main question is, should the embedded `TaskSpec` in a `PipelineSpec` have the exact same feature fields as a normal `TaskSpec` ? Should we have `Params` at all in the embedded `TaskSpec` at all ? (from a user perspective, we don't really need to care about the implementation). 

As a user, if I use an embedded `TaskSpec`, I kinda expect that the `PipelineSpec.params` can be used "as is" in the rest of the struct, which is not the case with the current state. I'm wondering if this is what others would expect or not :angel:

Reply:
oh i see.  Yeah I like that.  Pipeline has params but embedded task doesnt need them and just defers to the pipeline.

Implementation of that could be quite simple - embedded task spec is "hydrated" with a copy of the params from the pipeline before the taskrun is started.

Reply:
added couple of more variations just for users to understand it until we change the existing behavior.